### PR TITLE
Remove Newtonsoft

### PR DESCRIPTION
**Pull request by jtkech**
The biggest breaking change I never did ;)

For now more like a learning branch not ready to be reviewed.

Nothing works but at least the web app builds (not yet the tests).

### CODE REVIEW

**Comment by hishamco** [comment]
@jtkech me & @DAud-Ic planned to migrate to STJ https://github.com/OrchardCMS/OrchardCore/issues/14456

Personally, I made some progress in the PR https://github.com/OrchardCMS/OrchardCore/pull/14457, there is a local changes that not pushed yet, so we don't duplicate the work or wast the time for both of us, unless you are working on something related :)

**Comment by DAud-IcI** [comment]
> @jtkech me & @DAud-IcI planned to migrate to STJ #14456
> 
> Personally, I made some progress in the PR #14457, there is a local changes that have not pushed yet, so we don't duplicate the work or waste time for both of us unless you are working on something related :)

Yes, I too have made some progress building on @hishamco 's PR in this in the last few weeks. Just forgot to make a draft PR till now, but here it is https://github.com/OrchardCMS/OrchardCore/pull/14609

**Comment by hishamco** [comment]
I also made some progress, but seems JT keeps committing without checking our comments :)

**Comment by jtkech** [comment]
No problem, as mentioned this branch was more like a learning branch, but for now I have no time to continue, I will when I will have time.

That said the full Blog setup passes, and quite all unit tests unless some related to GraphQl and those related to Deployment that is not yet converted to SJT.

The main remaining thing is how to make `JsonObject` working as `dynamic` as it is used for example with Shapes and GraphQl arguments, but maybe better to remove all our usage of `dynamic` things.

**Comment by sebastienros** [comment]
@jtkech Look at my branch in yessql for how to get it to work with dynamics. There is a special converter based on the returned type.

**Comment by jtkech** [comment]
@sebastienros Thanks, yes I already tried it but even if it returns an `ExpandoObject` it was not fixing the GraphQl arguments, I will retry asap but this night I need to focus on this issue #14574.

**Comment by jtkech** [comment]
@sebastienros 

Yes the dynamic converter works, It needed to be added somewhere else.

I made it simpler by starting with the following, but for perf you may prefer your version.

        if (reader.TokenType != JsonTokenType.StartObject)
        {
            return JsonSerializer.Deserialize(ref reader, typeToConvert, options);
        }

        var properties = new ExpandoObject();
        while (reader.Read())
        {

**Comment by hishamco** [comment]
@jtkech if you will continue working on this we might need to close other PRs to avoid duplicating some of the works

Thanks for your HUGE efforts again!!

File: src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JsonDynamicValue.cs
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JsonDynamicValue.cs:20]:
```suggestion
    public object? Value => _value ??= JsonValue.ToObject<object>();
```

That what you meant?

Comment (jtkech) [reply src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JsonDynamicValue.cs:20]: Yes thanks, but `ToObject()` may return null so locally I now use an `_initialized` bool.

**Comment by Skrypt** [comment]
Wow, thanks for that work @jtkech

**Comment by jtkech** [comment]
@Skrypt No problem

The only problem is that now I'am in late for all other things ;)

Thanks for the encouragement.

**Comment by hishamco** [comment]
I expect this PR should merged in the upcoming major release because there are many breaking changes, right?

Either you will break other pending PRs, or other PR will breaks you all the time :)

**Comment by jtkech** [comment]
@MikeAlhayek Just for info.

When you get a json document on parsing it has to be disposed, but most of the time when parsing a not so big json string, better to use json node methods/helpers that cleanup things for us.

So here I updated your last changes in`ElasticSettingsDisplayDriver` and removed the new `JsonHelpers`.

**Comment by MikeAlhayek** [comment]
@jtkech Good to know. Thank you! Is this PR close to completion? I feel bad so many conflicts that keep occurring here that you have to resolve.

**Comment by MikeAlhayek** [comment]
@jtkech there are more conflicts here.

This PR has lots of changes. Is there anything outstanding that is preventing merging it? We should complete it and merge it soon to avoid anymore conflicts.

**Comment by jtkech** [comment]
Yes I agree, we should define a little plan.

- Do we want it, I think so, but we should define when with @sebastienros.

- When it's time, solve conflicts, re-focus on it to remember the basic tests to do.

- Then I think we should not merge it just before a new release, we need time to let people trying it because I think there will be some issues (hope not big ones) to fix.

- Then can be merged as is because we configure YesSql with an adapted serializer, as @sebastienros  did in a PR on YesSql side where I added some comments because here it is not exactly the same.
https://github.com/sebastienros/yessql/pull/504

**Comment by MikeAlhayek** [comment]
@jtkech @sebastienros Here is one way to do it
- Merge @sebastienros [PR](https://github.com/sebastienros/yessql/pull/504) in YesSql and release YesSql 4.0.0-beta-0004
- In this PR, update YesSql to 4.0.0-beta-0004
- Ask some people to test out this PR. I hope Lombiq guys can help test here @Piedone. The more we test the better the outcome. Maybe even release a special pre-release version on cloud smith (without merging) like (1.8.0-test-{build}) so it can be tested in other projects for wider tests.
- Once few people test it, we can release YesSql 4.0.0, update YesSql in this PR, and finally merge it. It could be merged around the first week of the new year assuming everything is good.
- Plan to release 1.8 end of January or early February.

**Comment by Piedone** [comment]
That sounds good. We'd definitely need a NuGet of this to be able to test:

- Orchard Core Commerce @sarahelsaig
- [OSOCE](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions) 
- DotNest @wAsnk 
- Lombiq websites @DemeSzabolcs 

Just let me know when a package is available.

Also, be sure to run the functional tests for this branch when ready too; I just started them [here](https://github.com/OrchardCMS/OrchardCore/actions/runs/7290212565).

**Comment by MikeAlhayek** [comment]
@Piedone in YesSql 4.0 we introduced breaking changes. And I think the idea is to release yessql 4.0 (with its breaking changes) along with OC 1.8 which. 

If we wait until 1.9, we'll properly need to release yessql 5.0 at that time. We'll see what @sebastienros prefers. Personally, I don't know if there is a rush for releasing 1.8. So maybe finish this and when all is done and tested we can release 1.8.

**Comment by sebastienros** [comment]
So we have an agreement for:
- Ship 1.8 with all the things we can in YesSQL 4.0 (including what @jtkech thiks we need to add to it). That will already imply a lot of breaking changes.
- As soon as 1.8 is released merge `System.Text.Json` so we can flush out all the migration issues for 1.9.

**Comment by Piedone** [comment]
Yeah, I'd do that. One set of impactful breaking changes is enough for a release.

It's not that there's a large rush for 1.8 but after .NET 8 I wouldn't wait for the JSON changes. And we'll have some good features for 1.9 too (I'm particularly awaiting https://github.com/OrchardCMS/OrchardCore/pull/14749 and https://github.com/OrchardCMS/OrchardCore/pull/7891).

**Comment by sebastienros** [comment]
I forgot that YesSql with STJ is another breaking change that can't go with STJ in OC. So we'll need a YesSql 5.0 actually with these changes.

**Comment by MikeAlhayek** [comment]
@jtkech I resolved the last set of conflicts. I also update this PR to use `YesSql 5.0.0-beta-0001` which contains [ Use System.Text.Json as the default serializer](https://github.com/sebastienros/yessql/pull/504)

**Comment by jtkech** [comment]
@MikeAlhayek 

> I resolved the last set of conflicts.

Thanks but in the commit comment I don't see the list of resolved conflicts, do you remember the files that you needed to update (I would want to check them) and/or if the conflicts were related to JSON?

> I also update this PR to use YesSql 5.0.0-beta-0001 which contains the STJ default serializer

Olay but as mentioned we first need to update this serializer before not overriding it here.

**Comment by MikeAlhayek** [comment]
Look at the last few commits I pushed.

Maybe this is what you are looking for? 

https://github.com/OrchardCMS/OrchardCore/commit/2caf92b02944609fe2bbfe6bb187ecc5b01b7475

Also, I hope we can release 1.8 soon (possible tomorrow if I hear back from Sebastien.) hopefully you'll have time to complete this PR and merge it to main after the release to avoid all these conflicts.

**Comment by jtkech** [comment]
> Maybe this is what you are looking for?

No, only the files that were in conflicts that normally are listed in the commit description, not the hundreds of the whole commit, do you remember if some were doing new things related to JSON?

Otherwise yolo ;)

**Comment by MikeAlhayek** [comment]
No it was not json related. But it was not many files. I know the Startup file of Elasticsearch had a conflict. Sorry I don't recall the rest of the files

**Comment by jtkech** [comment]
Okay, no problem if they were not json related

**Comment by hishamco** [comment]
@Piedone what's the status of this PR, can @sarahelsaig continue working on this or can I help on this, while me and @sarahelsaig started related PRs before JT started this

**Comment by MikeAlhayek** [comment]
@hishamco I'll work on this PR soon. I have to review the changes and then we'll need multiple people to test it out.

**Comment by hishamco** [comment]
I think there's some work-related on YesSQL to be done there

**Comment by MikeAlhayek** [comment]
@sebastienros 

Last comment JT stated the following
 > first need to update this serializer before not overriding it here.

I am guessing that comment is about [DefaultJsonContentSerializer.cs](https://github.com/OrchardCMS/OrchardCore/pull/14572/files#diff-0465a1aefa4bf27621e2b712f9a9926d869d68491f8cab5ffca939a98676cabf)

He seams to have replaced PoolingJsonContentSerializer with DefaultJsonContentSerializer. This replaces the YesSQL default implementation.

File: src/OrchardCore/OrchardCore.Recipes.Core/Services/RecipeExecutor.cs
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Recipes.Core/Services/RecipeExecutor.cs:56]: We need to check the history of this file. We may have changed it to read the document piece by piece for perf reasons.

File: src/OrchardCore/OrchardCore.Abstractions/Json/Serialization/ServiceCollectionExtensions.cs
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Abstractions/Json/Serialization/ServiceCollectionExtensions.cs:12]: This should be transient. Cheaper than singleton (won't be stored in root container), and is resolved once during the app (on startup).

**Comment by sebastienros** [comment]
We need to find all the collections and properties that are serialized in JSON and not declared using AddJsonDerivedType yet. Example: DeploymentStep. => throw an exception when the type is not found in JsonPolymorphicResolver.GeTypeInfo?

**Comment by MikeAlhayek** [comment]
@sebastienros I fixed latest conflicts. I hope you'll be able to finish off this PR. One change is that we now have `.AddDeployment<>` extensions for registering the deployment source, step and driver. So if you want to test adding a resolver for the deployment steps, you should be able to do that in one place.

**Comment by MikeAlhayek** [comment]
@sebastienros I fixed couple issues I ran into. But There is other problem I am running into.

There is an exception where setting `DeploymentPlan.Id` is throwing an exception.

```
  This exception was originally thrown at this call stack:
    OrchardCore.Deployment.DeploymentPlan.Id.set(long) in DeploymentPlan.cs
    YesSql.Data.PropertyAccessorFactory.LongAccessor<T>.YesSql.IAccessor<long>.Set(object, long) in PropertyAccessorFactory.cs
    YesSql.Session.Get<T>(System.Collections.Generic.IList<YesSql.Document>, string) in Session.cs
    YesSql.Services.DefaultQuery.Query<T>.ListImpl() in DefaultQuery.cs
    YesSql.Services.DefaultQuery.Query<T>.ListImpl() in DefaultQuery.cs
    OrchardCore.Deployment.DeploymentPlanService.GetDeploymentPlans() in DeploymentPlanService.cs
    OrchardCore.Deployment.DeploymentPlanService.GetAllDeploymentPlansAsync() in DeploymentPlanService.cs
    AspNetCoreGeneratedDocument.Views_Items_AddToDeploymentPlan_ActionDeploymentPlan_Modal.ExecuteAsync() in AddToDeploymentPlan-ActionDeploymentPlan.Modal.cshtml
    Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(Microsoft.AspNetCore.Mvc.Razor.IRazorPage, Microsoft.AspNetCore.Mvc.Rendering.ViewContext) in RazorView.cs
    Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageAsync(Microsoft.AspNetCore.Mvc.Razor.IRazorPage, Microsoft.AspNetCore.Mvc.Rendering.ViewContext, bool) in RazorView.cs
    ...
    [Call Stack Truncated]

```

Also, in the `PlaceholderAdminNodeNavigationBuilder` 58 is throwing an exception because line 57 is null.

You should be able to reproduce the issue by setting up a site using the blog recipe while on the main branch. Then checkout this branch and you should see the errors.

**Comment by MikeAlhayek** [comment]
Here is one issue. The content serializers can't deserialize this json

It can't desalinize these object type 

```
OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep`1[[OrchardCore.AuditTrail.Settings.AuditTrailSettings, OrchardCore.AuditTrail]], OrchardCore.Settings.Core

OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep`1[[OrchardCore.Admin.Models.AdminSettings, OrchardCore.Admin.Abstractions]], OrchardCore.Settings.Cor
```

Here is the full json
```
{
    "Id":49,
    "Name":"Export content to deployment plan target",
    "DeploymentSteps":[
        {
           "$type":"OrchardCore.Contents.Deployment.ExportContentToDeploymentTarget.ExportContentToDeploymentTargetDeploymentStep, OrchardCore.Contents",
            "Id":"4qzdq3pnzvas5v6mxvc17g8em9",
            "Name":"ExportContentToDeploymentTargetDeploymentStep"
        },
        {
            "$type":"OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep`1[[OrchardCore.AuditTrail.Settings.AuditTrailSettings, OrchardCore.AuditTrail]], OrchardCore.Settings.Core",
            "Id":"a2e78891bc6e48fc8eda3bf3b69922fb",
            "Name":"AuditTrailSettings_SiteSettingsPropertyDeploymentStep"
        },
        {
            "$type":"OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep`1[[OrchardCore.Admin.Models.AdminSettings, OrchardCore.Admin.Abstractions]], OrchardCore.Settings.Core",
            "Id":"dba2b359d044495f86f320e58ada7319",
            "Name":"AdminSettings_SiteSettingsPropertyDeploymentStep"
        }
    ]
}
```

we already added 

```
services.AddJsonDerivedTypeInfo<SiteSettingsPropertyDeploymentStep<TModel>, DeploymentStep>();
```

which works when the deployment step looks like this but not when the steps are settings as shown above.

```
{
    "Id":18,
    "Name":"test",
    "DeploymentSteps":[
        {
            "$type":"OrchardCore.ContentTypes.Deployment.ContentDefinitionDeploymentStep, OrchardCore.ContentTypes",
            "IncludeAll":false,
            "ContentTypes":[
                "Test"
            ],
            "ContentParts":[
                "Test"
            ],
            "Id":"4d154596f6af4cfa88fc72569027bbe5",
            "Name":"ContentDefinition"
        }
    ]
}
```

**Comment by sebastienros** [comment]
I fixed the admin nodes / menu items.

**Comment by sebastienros** [comment]
> Here is one issue. The content serializers can't deserialize this json

Worked for me. I was able to add an Audit Trail Settings step to the plan and reload it.
Or did you actually create it with main then loaded it with the branch?

**Comment by MikeAlhayek** [comment]
Yes. I created a site from main branch and the use this branch to test. 

Maybe update site settings and then try to export the site settings

**Comment by sebastienros** [comment]
I will assume that's what you tried, since you mention 
```
"$type":"OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep`1[[OrchardCore.AuditTrail.Settings.AuditTrailSettings, OrchardCore.AuditTrail]], OrchardCore.Settings.Core"
```

and with this cranch I see 

```
"$type": "OrchardCore.Settings.Deployment.SiteSettingsPropertyDeploymentStep\u00601[[OrchardCore.Contents.AuditTrail.Settings.ContentAuditTrailSettings, OrchardCore.Contents, Version=1.9.0.0, Culture=neutral, PublicKeyToken=null]], OrchardCore.Settings.Core"
```

I am fixing it by importing the logic from NewtonSoft.Json.

**Comment by sebastienros** [comment]
The issue on the long is probably due to that, since it can't deserialize the document is returns null to YesSql and then when `IIdAccessor.Set` is called it's a null document and it fails. You should try again with my fix, it should work.

**Comment by MikeAlhayek** [comment]
@sebastienros I am running into another issue

The migrations are failing. 

> Error while running migration version 0 for 'OrchardCore.Seo'. System.InvalidOperationException: The node already has a parent.

```
2024-01-31 15:44:25.1676|jt3|00-f946c032ab542635ddb52676058ac31c-4900de439ad45011-00||OrchardCore.Data.Migration.DataMigrationManager|ERROR|Error while running migration version 0 for 'OrchardCore.Seo'. System.InvalidOperationException: The node already has a parent.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_NodeAlreadyHasParent()
   at System.Text.Json.Nodes.JsonNode.AssignParent(JsonNode parent)
   at System.Text.Json.Nodes.JsonObject.SetItem(String propertyName, JsonNode value)
   at OrchardCore.CustomSettings.Recipes.CustomSettingsStep.ExecuteAsync(RecipeExecutionContext context) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore.Modules\OrchardCore.CustomSettings\Recipes\CustomSettingsStep.cs:line 38
   at OrchardCore.Recipes.Services.RecipeExecutor.<>c__DisplayClass7_0.<<ExecuteStepAsync>b__0>d.MoveNext() in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 173
--- End of stack trace from previous location ---
   at OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func`2 execute, Boolean activateShell) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Abstractions\Shell\Scope\ShellScope.cs:line 237
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteStepAsync(RecipeExecutionContext recipeStep) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 156
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, IDictionary`2 environment, CancellationToken cancellationToken) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 99
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, IDictionary`2 environment, CancellationToken cancellationToken) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 117
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, IDictionary`2 environment, CancellationToken cancellationToken) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 132
   at OrchardCore.Recipes.Services.RecipeExecutor.ExecuteAsync(String executionId, RecipeDescriptor recipeDescriptor, IDictionary`2 environment, CancellationToken cancellationToken) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeExecutor.cs:line 142
   at OrchardCore.Recipes.Services.RecipeMigrator.ExecuteAsync(String recipeFileName, IDataMigration migration) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Recipes.Core\Services\RecipeMigrator.cs:line 55
```

**Review by sebastienros — APPROVED** [review]

**Comment by MikeAlhayek** [comment]
@sebastienros there is one more issue.

```
has occurred while executing the request. System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at YesSql.Indexes.IdentityMap.AddEntity(Int64 id, Object entity)
   at YesSql.Session.Get[T](IList`1 documents, String collection)
   at YesSql.Services.DefaultQuery.Query`1.FirstOrDefaultImpl()
   at YesSql.Services.DefaultQuery.Query`1.FirstOrDefaultImpl()
   at OrchardCore.Data.Documents.DocumentStore.GetOrCreateImmutableAsync[T](Func`1 factoryAsync) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Data.YesSql\Documents\DocumentStore.cs:line 56
   at OrchardCore.Documents.DocumentManager`1.GetOrCreateImmutableAsync(Func`1 factoryAsync) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore\OrchardCore.Infrastructure\Documents\DocumentManager.cs:line 143
   at OrchardCore.Layers.Services.LayerFilter.OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next) in C:\Code\OrchardCMS\OrchardCore\src\OrchardCore.Modules\OrchardCore.Layers\Services\LayerFilter.cs:line 91
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|30_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

**Comment by MikeAlhayek** [comment]
@sebastienros the last error is strange. After digging little more here is the error that I am getting

```
Deserialization of types without a parameterless constructor, a singular parameterized constructor, or a parameterized constructor annotated with 'JsonConstructorAttribute' is not supported. Type 'OrchardCore.Rules.Models.StringOperator'. Path: $.Layers[2].LayerRule.Conditions[0].Operation | LineNumber: 0 | BytePositionInLine: 669.'
```

the `StringOperator` has parameter-less constructor. and `$.Layers[2].LayerRule.Conditions[0].Operation`

Here is the layer JSON from the database

```
{
    "Layers":[
        {
            "Name":"Knowledge Base",
            "LayerRule":{
                "Conditions":[
                    {
                        "$type":"OrchardCore.Rules.Models.AnyConditionGroup, OrchardCore.Rules",
                        "Conditions":[
                            {
                                "$type":"OrchardCore.Rules.Models.UrlCondition, OrchardCore.Rules",
                                "Value":"/knowledge-base/category/",
                                "Operation":{
                                    "$type":"OrchardCore.Rules.Models.StringStartsWithOperator, OrchardCore.Rules",
                                    "CaseSensitive":false
                                },
                                "Name":"UrlCondition",
                                "ConditionId":"4q7ayd91z69gzv8qzta07rtpm0"
                            },
                            {
                                "$type":"OrchardCore.Rules.Models.ContentTypeCondition, OrchardCore.Rules",
                                "Value":"KnowledgeBaseArticle",
                                "Operation":{
                                    "$type":"OrchardCore.Rules.Models.StringEqualsOperator, OrchardCore.Rules",
                                    "CaseSensitive":false
                                },
                                "Name":"ContentTypeCondition",
                                "ConditionId":"4bee24643tz0nsgwn948venadz"
                            },
                            {
                                "$type":"OrchardCore.Rules.Models.RoleCondition, OrchardCore.Rules",
                                "Value":"/search",
                                "Operation":{
                                    "$type":"OrchardCore.Rules.Models.StringStartsWithOperator, OrchardCore.Rules",
                                    "CaseSensitive":false
                                },
                                "Name":"RoleCondition",
                                "ConditionId":"4017ez2rhw7za5fxmcn0kgbw5k"
                            }
                        ],
                        "Name":"AnyConditionGroup",
                        "ConditionId":"49kzd1py7h3vnx29608cmmta5a"
                    }
                ],
                "ConditionId":"4jm7m68aj30g0rczdne1c4b1cp"
            }
        },
        {
            "Name":"Knowledge Base Documents",
            "LayerRule":{
                "Conditions":[
                    {
                        "$type":"OrchardCore.Rules.Models.AnyConditionGroup, OrchardCore.Rules",
                        "Conditions":[
                            {
                                "$type":"OrchardCore.Rules.Models.UrlCondition, OrchardCore.Rules",
                                "Value":"/knowledge-base/category/",
                                "Operation":{
                                    "$type":"OrchardCore.Rules.Models.StringStartsWithOperator, OrchardCore.Rules",
                                    "CaseSensitive":false
                                },
                                "Name":"UrlCondition",
                                "ConditionId":"4rgsspd31y3m67jk68q0ny57ng"
                            },
                            {
                                "$type":"OrchardCore.Rules.Models.ContentTypeCondition, OrchardCore.Rules",
                                "Value":"KnowledgeBaseArticle",
                                "Operation":{
                                    "$type":"OrchardCore.Rules.Models.StringEqualsOperator, OrchardCore.Rules",
                                    "CaseSensitive":false
                                },
                                "Name":"ContentTypeCondition",
                                "ConditionId":"4s8zc1yn4qg64xscsdth3c6jcf"
                            }
                        ],
                        "Name":"AnyConditionGroup",
                        "ConditionId":"4jd36bg0n6p2htsvrw5vgf6cdt"
                    }
                ],
                "ConditionId":"4ywdwwjm06x2t2t0yr332en2wd"
            }
        },
        {
            "Name":"Home Page",
            "Description":"A layer to make content visible on home page.",
            "LayerRule":{
                "Conditions":[
                    {
                        "$type":"OrchardCore.Rules.Models.HomepageCondition, OrchardCore.Rules",
                        "Value":true,
                        "Name":"HomepageCondition",
                        "ConditionId":"4aqqr1zz96va9s6b3nhy9afvw6"
                    }
                ],
                "ConditionId":"4kpbhnws6dm14yxyc0rt4phw5p"
            }
        },
        {
            "Name":"Secure Home Page",
            "Description":"A layer to make content visible on home page when the user is authenticated.",
            "LayerRule":{
                "Conditions":[
                    {
                        "$type":"OrchardCore.Rules.Models.IsAuthenticatedCondition, OrchardCore.Rules",
                        "Name":"IsAuthenticatedCondition",
                        "ConditionId":"4cw5t50rfyvm61agcs1xs2v6x6"
                    },
                    {
                        "$type":"OrchardCore.Rules.Models.HomepageCondition, OrchardCore.Rules",
                        "Value":true,
                        "Name":"HomepageCondition",
                        "ConditionId":"4jm3g93fvqg3k6ese610vtvzcf"
                    }
                ],
                "ConditionId":"4jvy81v5dvem66dk43ysqr64xv"
            }
        },
        {
            "Name":"Anonymous Home Page",
            "Description":"A widget layer for unauthenticated users.",
            "LayerRule":{
                "Conditions":[
                    {
                        "$type":"OrchardCore.Rules.Models.HomepageCondition, OrchardCore.Rules",
                        "Value":true,
                        "Name":"HomepageCondition",
                        "ConditionId":"4b14hdp7q60xw63ybp0q2wedxt"
                    },
                    {
                        "$type":"OrchardCore.Rules.Models.IsAnonymousCondition, OrchardCore.Rules",
                        "Name":"IsAnonymousCondition",
                        "ConditionId":"4tcmmkf7mzsjd41awsgf3bjy5s"
                    }
                ],
                "ConditionId":"4trw79793xqg4r1hkdfv6s83dj"
            }
        }
    ],
    "Identifier":"4cy4fpeft8xsywx7f1v6w2bjdt"
}
```

**Comment by sebastienros** [comment]
The Rules issue is because there is a hierarchy to annotate. I am fixing it, using a different way than the services though.

**Comment by MikeAlhayek** [comment]
@sebastienros thank you! that fixed the last issues I was able to spot. Can you please look at the release notes to make sure I did not miss anything that needs to be done in order to upgrade?

https://github.com/OrchardCMS/OrchardCore/blob/b3a9ab4ae5b07d06a5eb1729dd0ef1fd0d633a3f/src/docs/releases/1.9.0.md

**Review by MikeAlhayek — APPROVED** [review]
Thank you @jtkech! You are still contributing my friend!

**Comment by sebastienros** [comment]
I have changed it to use options instead of the DI. It still worked on my machine after (tried deployments and layer rules). The goal was to have less things in DI.

**Comment by sebastienros** [comment]
I started a functional tests run on this branch before we merge.

**Comment by sebastienros** [comment]
Updated release notes, functional tests passed.

**Comment by MikeAlhayek** [comment]
@Piedone This beast was just merged. Can you please use dotnest or other ways to test it out to make sure we did not break anything?

cc: @hishamco @Skrypt @agriffard

**Comment by hishamco** [comment]
Do we have a release note, because this is HUGE!!

**Comment by MikeAlhayek** [comment]
yes it should be part of the 1.9 release notes. I think it should be published soon. guessing it take a while for the docs to be published

**Comment by Piedone** [comment]
We'll first need to finish [the 1.8 upgrade](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/638) but after that we'll look into this: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/692. Please wait with an 1.9 release for that because that can bring up a lot of feedback.

**Pull request merged**
